pkg/consensus/consensus.go

```diff
@@ -296,6 +296,7 @@ func (s *service) Shutdown() {
 			s.wallet.Close()
 		}
 	}
+	_ = s.log.Sync()
```

Comment:
Edit the commit message, please. This commit won't fix the data race itself since data race is caused by missing synchronization channels in the network server. This commit is only an enhancement, it makes the code follow the zap usages rules:
```
// Sync calls the underlying Core's Sync method, flushing any buffered log
// entries. Applications should take care to call Sync before exiting.
```
You may even insert this citation in the commit message.

It's the next commit (`*Fin` channels addition) who should fix two issues: #2973 and #2974. And it's a separate commit for #3112 that is described in https://github.com/nspcc-dev/neo-go/pull/3307#discussion_r1492445285.